File: YoutubeLinks.Shared/Features/Playlists/Commands/ExportPlaylist.cs
using FluentValidation;
using MediatR;
using Microsoft.Extensions.Localization;
using YoutubeLinks.Shared.Features.Playlists.Helpers;
using YoutubeLinks.Shared.Localization;

namespace YoutubeLinks.Shared.Features.Playlists.Commands;

public static class ExportPlaylist
{
    public class Command : IRequest<PlaylistFile>
    {
        public int Id { get; init; }
        public PlaylistFileType PlaylistFileType { get; init; }
    }

    public class Validator : AbstractValidator<Command>
    {
        public Validator(IStringLocalizer<ValidationMessage> localizer)
        {
            RuleFor(x => x.PlaylistFileType)
                .IsInEnum()
                .WithMessage(x => localizer[nameof(ValidationMessageString.PlaylistFileTypeIsInEnum)]);
        }
    }
}

File: youtube-links/src/shared/features/playlists/commands/ExportPlaylist.ts
import { isValidPlaylistFileTypeValue, type PlaylistFileType } from '../helpers/PlaylistFileType';

export namespace ExportPlaylist {
  export interface Command {
    id: number;
    playlistFileType: PlaylistFileType;
  }

  export const Validation = {
    playlistFileType: {
      isInEnum: (v: string) =>
        isValidPlaylistFileTypeValue(v) ||
        `PlaylistFileType has a range of values which does not include: ${v}.`,
    },
  };
}
